src/plugins/Library/ArchiverFactory.java

```diff
@@ -0,0 +1,18 @@
+package plugins.Library;
+
+import plugins.Library.io.ObjectStreamReader;
+import plugins.Library.io.ObjectStreamWriter;
+import plugins.Library.io.serial.LiveArchiver;
+import plugins.Library.util.exec.SimpleProgress;
+
+public interface ArchiverFactory {
+    <T, S extends ObjectStreamWriter & ObjectStreamReader>
+		  LiveArchiver<T, SimpleProgress>
+		  newArchiver(S rw, String mime, int size,
```

Comment:
Methods in interfaces should have at least rudimentary documentation, otherwise nobody will ever know what that `int` is meant to be…